platform-server-kernel/src/main/java/uno/cod/platform/server/core/service/InvitationService.java

```diff
@@ -0,0 +1,103 @@
+package uno.cod.platform.server.core.service;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+import uno.cod.platform.server.core.dto.challenge.ChallengeShowDto;
+import uno.cod.platform.server.core.dto.invitation.InvitationDto;
+import uno.cod.platform.server.core.dto.organization.OrganizationShowDto;
+import uno.cod.platform.server.core.dto.user.UserCreateDto;
+import uno.cod.platform.server.core.dto.user.UserShowDto;
+
+import javax.mail.MessagingException;
+import javax.mail.Transport;
+import javax.mail.internet.MimeMessage;
+import java.util.Arrays;
+
+@Service
+public class InvitationService {
+    private final UserService userService;
+    private final OrganizationService organizationService;
+    private final ChallengeService challengeService;
+    private final Environment environment;
+    private Logger log = Logger.getLogger(InvitationService.class.getName());
+
+    @Autowired
+    public InvitationService(UserService userService, OrganizationService organizationService, ChallengeService challengeService, Environment environment) {
+        this.userService = userService;
+        this.organizationService = organizationService;
+        this.challengeService = challengeService;
+        this.environment = environment;
+    }
+
+    public void invite(InvitationDto dto) {
+        OrganizationShowDto organization = organizationService.findById(dto.getOrganizationId());
+        if (organization == null) {
+            throw new IllegalArgumentException("invalid organization");
+        }
+        ChallengeShowDto challenge = challengeService.findById(dto.getChallengeId());
+        if (challenge == null) {
+            throw new IllegalArgumentException("invalid challenge");
+        }
+        UserShowDto user = userService.findByEmail(dto.getEmail());
+        if (user == null) {
+            UserCreateDto userCreateDto = new UserCreateDto();
+            userCreateDto.setEmail(dto.getEmail());
+            userCreateDto.setNick(dto.getEmail());
+            userCreateDto.setPassword("random_password");
+            userService.createFromDto(userCreateDto);
+        }
+        user = userService.findByEmail(dto.getEmail());
+        String mailMessage = getChallengeStartMailMessage(user, organization, challenge);
+
+        if (Arrays.asList(this.environment.getActiveProfiles()).contains("development")) {
+            log.log(Level.ERROR, mailMessage);
+        } else {
+            // TODO - send mail
+            MimeMessage message = null;
+            try {
+                Transport.send(message);
+            } catch (MessagingException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private String getChallengeStartMailMessage(UserShowDto user, OrganizationShowDto organization, ChallengeShowDto challenge) {
```

Comment:
=> invite.html on master is already the stub for that, use thymeleaf templating language, at some point we may prettify the invation mail ... maybe we should do it right now